tests/Tokenizer/Transformer/TypeAlternationTransformerTest.php

```diff
@@ -89,6 +89,32 @@ function foo(){}
                 ',
             ],
         ];
+
+        yield 'self as type' => [
+            '<?php class Foo {
+                function f1(bool|self|int $x): void {}
+                function f2(): self|\stdClass {}
+            }',
+            [
+                12 => CT::T_TYPE_ALTERNATION,
+                14 => CT::T_TYPE_ALTERNATION,
+                34 => CT::T_TYPE_ALTERNATION,
+            ],
+        ];
+
+        yield 'static as type' => [
+            '<?php class Foo {
+                function f1(): static|TypeA {}
```

Comment:
```suggestion
                function f1(static|TypeA): void {}
```
Maybe like for `self`?

Reply:
we may want to test both, param type and return type

Reply:
Sorry for the noise, I was wrong. `static` [can be only used as return type](https://wiki.php.net/rfc/static_return_type).